src/assets/CollectiblesController.test.ts

```diff
@@ -598,18 +685,29 @@ describe('CollectiblesController', () => {
 
   describe('RemoveCollectible', () => {
     it('should remove collectible and collectible contract', async () => {
+      const { selectedAddress, chainId } = collectiblesController.config;
+
       await collectiblesController.addCollectible('0x01', '1', {
         name: 'name',
         image: 'image',
         description: 'description',
         standard: 'standard',
       });
       collectiblesController.removeCollectible('0x01', '1');
-      expect(collectiblesController.state.collectibles).toHaveLength(0);
-      expect(collectiblesController.state.collectibleContracts).toHaveLength(0);
+      expect(
+        collectiblesController.state.allCollectibles[selectedAddress][chainId],
+      ).toHaveLength(0);
+
+      expect(
+        collectiblesController.state.allCollectibleContracts[selectedAddress][
+          chainId
+        ],
+      ).toHaveLength(0);
```

Comment:
Should we remove the key rather than just remove the item from the array when there are no more left for the address/chainId combo?

Reply:
I agree that this would be nicer, so that the behaviour after adding and removing a collectible matches the initial behaviour. Leaves us with less cases to test, and it's simpler for anyone using this state. I don't want to block on it though.